From 8a073cdd39b10cb6b21aebea2554749d98c87195 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 30 May 2023 22:01:00 +0530
Subject: [PATCH] Resolved compilation issue for media driver

Changes include:
- Added Android flag to disable error mode similar to Linux
- Added header file for errno
- Added include path for libva and cmrtlib
- Removed files resulting in duplicate symbol

Change-Id: Ia54ad1d036783dabca1b18c2bade9f882b56a8a0
Tracked-On: OAM-106860
Signed-off-by: Ratnesh Kumar Rai <ratnesh.kumar.rai@intel.com>
Signed-off-by: yerriswamy <yerriswamy.kt@intel.com>
---
 cmrtlib/linux/hardware/cm_device_os.cpp                   | 6 ++++--
 media_driver/Android.mk                                   | 8 ++------
 .../agnostic/common/vp/kdll/hal_kerneldll_next.c          | 7 ++++---
 .../linux/Xe_M_plus/ddi/media_libva_caps_mtl_base.cpp     | 1 +
 media_softlet/linux/common/os/i915/include/mos_bufmgr.h   | 1 +
 .../linux/common/os/osservice/mos_utilities_specific.cpp  | 2 ++
 6 files changed, 14 insertions(+), 11 deletions(-)

diff --git a/cmrtlib/linux/hardware/cm_device_os.cpp b/cmrtlib/linux/hardware/cm_device_os.cpp
index 36b98c13a..d63221f8f 100644
--- a/cmrtlib/linux/hardware/cm_device_os.cpp
+++ b/cmrtlib/linux/hardware/cm_device_os.cpp
@@ -495,8 +495,10 @@ CmDevice_RT::CmDevice_RT(
     m_gtpinBufferUP1(nullptr),
     m_gtpinBufferUP2(nullptr),
     m_createOption(createOption),
-    m_driverStoreEnabled(0),
-    m_driFileDescriptor(0)
+    m_driverStoreEnabled(0)
+#ifndef ANDROID
+    , m_driFileDescriptor(0)
+#endif
 {
 
     // New Surface Manager
diff --git a/media_driver/Android.mk b/media_driver/Android.mk
index 8f6a618b5..463fae17f 100644
--- a/media_driver/Android.mk
+++ b/media_driver/Android.mk
@@ -430,7 +430,6 @@ LOCAL_SRC_FILES := \
     ../media_softlet/agnostic/common/os/mos_util_debug.cpp \
     ../media_softlet/agnostic/common/os/mos_utilities_inner.cpp \
     ../media_softlet/agnostic/common/os/mos_utilities_next.cpp \
-    ../media_softlet/agnostic/common/os/private/mos_utilities_usersetting.cpp \
     ../media_softlet/agnostic/common/os/share/mos_oca_util_debug.cpp \
     ../media_softlet/agnostic/common/os/user_setting/media_user_setting.cpp \
     ../media_softlet/agnostic/common/os/user_setting/media_user_setting_configure.cpp \
@@ -599,10 +598,6 @@ LOCAL_SRC_FILES := \
     ../media_softlet/linux/common/os/mos_vma.c \
     ../media_softlet/linux/common/os/osservice/mos_util_debug_specific.cpp \
     ../media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp \
-    ../media_softlet/linux/common/os/private/mos_decompression.cpp \
-    ../media_softlet/linux/common/os/private/mos_mediacopy.cpp \
-    ../media_softlet/linux/common/os/private/mos_os_specific.cpp \
-    ../media_softlet/linux/common/os/private/mos_utilities_specific_usersetting.cpp \
     ../media_softlet/linux/common/os/user_setting/media_user_setting_configure_specific.cpp \
     ../media_softlet/linux/common/shared/hal_oca_interface_next.cpp \
     ../media_softlet/linux/common/shared/skuwa_dumper_specific.c \
@@ -1981,7 +1976,8 @@ LOCAL_C_INCLUDES  = \
     $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/decode/vp8/ddi \
     $(LOCAL_PATH)/../media_softlet/linux/xe_lpm_plus_r0/decode/vp9/ddi \
     $(LOCAL_PATH)/../media_softlet/media_interface/media_interfaces_mtl \
-
+    $(LOCAL_PATH)/../../libva/va \
+    $(LOCAL_PATH)/../cmrtlib/linux/hardware
 
 #LOCAL_CPP_FEATURES := rtti exceptions
 
diff --git a/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c b/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c
index d8f7d429a..7d5ea5b7c 100644
--- a/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c
+++ b/media_softlet/agnostic/common/vp/kdll/hal_kerneldll_next.c
@@ -3918,6 +3918,7 @@ bool KernelDll_BuildKernel_CmFc(Kdll_State *pState, Kdll_SearchState *pSearchSta
     VP_RENDER_FUNCTION_ENTER;
 
     // Disable pop-up box window for STL assertion to avoid VM hang in auto test.
+#ifndef ANDROID
 #if (!LINUX)
     uint32_t prevErrorMode = ::SetErrorMode(SEM_FAILCRITICALERRORS | SEM_NOGPFAULTERRORBOX);
 #if defined(_MSC_VER)
@@ -3930,7 +3931,7 @@ bool KernelDll_BuildKernel_CmFc(Kdll_State *pState, Kdll_SearchState *pSearchSta
     _CrtSetReportFile(_CRT_ASSERT, _CRTDBG_FILE_STDERR);
 #endif
 #endif
-
+#endif
     pSearchState->KernelLink.dwSize  = DL_MAX_SYMBOLS;
     pSearchState->KernelLink.dwCount = 0;
     pSearchState->KernelLink.pLink   = pSearchState->LinkArray;
@@ -4048,7 +4049,7 @@ bool KernelDll_BuildKernel_CmFc(Kdll_State *pState, Kdll_SearchState *pSearchSta
     res = true;
 
 finish:
-#if (!LINUX)
+#if (!LINUX) && !defined(ANDROID)
     ::SetErrorMode(prevErrorMode);
 #endif
     return res;
@@ -4454,4 +4455,4 @@ bool KernelDll_SetupFunctionPointers_Ext(
 
 #ifdef __cplusplus
 }
-#endif  // __cplusplus
\ No newline at end of file
+#endif  // __cplusplus
diff --git a/media_softlet/linux/Xe_M_plus/ddi/media_libva_caps_mtl_base.cpp b/media_softlet/linux/Xe_M_plus/ddi/media_libva_caps_mtl_base.cpp
index 3949d5b55..715305585 100644
--- a/media_softlet/linux/Xe_M_plus/ddi/media_libva_caps_mtl_base.cpp
+++ b/media_softlet/linux/Xe_M_plus/ddi/media_libva_caps_mtl_base.cpp
@@ -37,6 +37,7 @@
 #include "media_ddi_decode_const_g12.h"
 #include "media_ddi_encode_const.h"
 #include "drm_fourcc.h"
+#include "va_android.h"
 
 const VAImageFormat m_supportedImageformatsXe_Lpm_Plus_Base[] =
 {   {VA_FOURCC_BGRA,           VA_LSB_FIRST,   32, 32, 0x0000ff00, 0x00ff0000, 0xff000000,  0x000000ff}, /* [31:0] B:G:R:A 8:8:8:8 little endian */
diff --git a/media_softlet/linux/common/os/i915/include/mos_bufmgr.h b/media_softlet/linux/common/os/i915/include/mos_bufmgr.h
index 9f2fd36ad..41fba3033 100644
--- a/media_softlet/linux/common/os/i915/include/mos_bufmgr.h
+++ b/media_softlet/linux/common/os/i915/include/mos_bufmgr.h
@@ -34,6 +34,7 @@
 #ifndef MOS_BUFMGR_H
 #define MOS_BUFMGR_H
 
+#include <errno.h>
 #include <stdio.h>
 #include <stdint.h>
 #include <stdio.h>
diff --git a/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp b/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
index 97ad15c62..6090a49aa 100644
--- a/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
+++ b/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
@@ -2542,6 +2542,7 @@ void MosUtilities::MosTraceEvent(
                 MOS_FreeMemory(pTraceBuf);
             }
         }
+#ifndef ANDROID
 #if Backtrace_FOUND
         if (m_mosTraceFilter(TR_KEY_CALL_STACK))
         {
@@ -2562,6 +2563,7 @@ void MosUtilities::MosTraceEvent(
                 size_t ret = write(MosUtilitiesSpecificNext::m_mosTraceFd, traceBuf, nLen);
             }
         }
+#endif
 #endif
     }
     return;
-- 
2.17.1

